runtime/gc_structs/ClassLoaderClassesIterator.cpp

```diff
@@ -108,13 +152,18 @@ GC_ClassLoaderClassesIterator::nextClass()
 		if (ANONYMOUS_CLASSES == _mode) {
 			_nextClass = nextAnonymousClass();
 		} else {
-			if ( (result->classLoader == _classLoader) && (NULL != result->arrayClass) ) {
+			J9Class* array = nextArrayClass();
```

Comment:
This should be `J9ArrayClass*` and `nextArrayClass()` should return `J9ArrayClass* `

Reply:
My preference is to have less casting, I don't think it matters.